sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+		exit_list = Map[target_room].wayto
+		exit_list.each do |rnum, dir|
+		  #flag to listen for room numbers appearing in 'peer' info
+		  Flags.add('sigil-room-detected', /Room Number: \d+/)
+		  DRC.bput("peer #{dir}", /You peer/, /You see/, /\[Usage:/)
+		  pause 0.5
+		  #if we detected a room nubmer, move in that direction and return
+		  if Flags['sigil-room-detected']
+		    DRC.retreat
+		    fput(way)
+			Flags.reset('sigil-room-detected')
+			return
+		  end
+		end
+		#If we are unable to find a mapped exit, write the original target room to nogo and disconnect as a safety measure
+		DRC.message('Cannot find a way out. Exiting script.')
+		add_to_nogo(target_room, @data)
+		exit
+	end #unstuck_me
+	
+	#uses 'perc sigil' repeatedly until both sigils are discovered
+	def find_sigils
+		room_sigils = []
+		while room_sigils.length < 2
+			#weave is a 0 RT way of detecting enemies in the room. It will auto face enemies but will not harass players or NPCs. After weaving check if the danger-room flag has tripped
+			DRC.bput('weave', /You weave back and forth in a dance-like motion/, /You must be closer to use tactical abilities on your opponent/, /There is nothing else to face/)
+			if Flags['sigil-danger-room']
+			  add_to_nogo(Room.current.id, @data) unless Room.current.id == nil
+			  return nil
+			end 
+
+			#main 'perc sigil' DRC.bput and case statement. When sigils are detected they're added to an array
+			case line = DRC.bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about|This is not an appropriate place for that/)
+			when /^You recall.*/
+			  DRC.message('Room searched too recently')
+			  return nil
+			when /You are too distracted/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /Something about|This is not an appropriate place for that/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /^Having recently been searched/
+			  return nil
+			when *@sigil_list
+			  room_sigils.push(line)
+			  DRC.message("Found: #{room_sigils}")
+			end
+		end #while
+
+		#return the array of all sigils discovered in the room (should be 2 unless something went wrong)
+		return room_sigils
+	end
+	
+	#generate the list of all rooms that contain the current room's Room.current.image data
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		#generate the array
+		roomlist = []
+		data.each do |room|
+		  roomlist.push(room.id)
+		end
+
+		#return the full list of rooms
+		return roomlist
+	end #region_room_list
+	
+	#writes a roomid to the nogo list which is subtracted from the walk_list when the script is run in the future
+	def add_to_nogo(banroom, file_data)
+		#initialize a new array if nogo is empty
+		if file_data.SigilInfo['nogo'] == nil
+		  file_data.SigilInfo['nogo'] = []
+		end
```

Comment:
💡 A ruby-ish way to do this is the conditional assignment: `file_data.SigilInfo['nogo'] ||= []`

If the variable is nil then it's assigned the right-hand side value, else the variable's value is not re-assigned.

Reply:
Noice. Implemented.

Reply:
Added!